lib/link/inventory.rb

```diff
@@ -0,0 +1,24 @@
+module Link
+  class Inventory
+    attr_reader :inventory, :user
+
+    delegate :network_partner?, to: :user
+    delegate :facilitator?, :participant?, to: :inventory
+
+    def initialize(inventory, user)
+      @inventory = inventory
+      @user = user
+    end
+
+    def execute
+      target.new(inventory, user).execute
+    end
+
+    private
+    def target
+      return Link::InventoryFacilitator if facilitator?(user: user)
+      return Link::InventoryParticipant if participant?(user: user)
+      Link::Partner
+    end
+  end
+end
```

Comment:
Functionality in `/lib` is conventionally reserved for cross-application library code, akin to gems. This should really be in `app/services` or similar.
I know this is the pattern we've been following, but thought I'd mention this anyway.